src/test/java/org/broadinstitute/consent/http/service/DatasetServiceTest.java

```diff
@@ -767,10 +768,46 @@ void testApproveDataset_DenyDataset() throws Exception {
     verify(emailService, times(1)).sendDatasetDeniedMessage(
         user,
         "DAC NAME",
-        "DUOS-000001"
+        "DUOS-000001",
+        "dacEmail@gmail.com"
     );
   }
 
+  @Test
+  void testApproveDataset_DenyDataset_WithNoDACEmail() throws Exception {
+    Dataset dataset = new Dataset();
+    dataset.setDataSetId(1);
+    User user = new User();
+    user.setUserId(1);
+    user.setEmail("asdf@gmail.com");
+    user.setDisplayName("John Doe");
+    Boolean payloadBool = false;
+    Dataset updatedDataset = new Dataset();
+    updatedDataset.setDataSetId(1);
+    updatedDataset.setDacApproval(payloadBool);
+
+    when(datasetDAO.findDatasetById(any())).thenReturn(updatedDataset);
+    dataset.setAlias(1);
+    dataset.setDacId(3);
+    Dac dac = new Dac();
+    dac.setName("DAC NAME");
+    initService();
+    when(dacDAO.findById(3)).thenReturn(dac);
+
+    Dataset returnedDataset = datasetService.approveDataset(dataset, user, payloadBool);
+    assertEquals(dataset.getDataSetId(), returnedDataset.getDataSetId());
+    assertFalse(returnedDataset.getDacApproval());
+
+    // do not send denied email
+    verify(emailService, times(0)).sendDatasetDeniedMessage(
```

Comment:
What's the expected behavior of this test? Where does email validation take place? Is the expected behavior the logged warning?

Reply:
I agree, I was expecting to see a test that would cause `emailService.sendDatasetDeniedMessage()` to be called as that's the new code here.

Reply:
Yes, in this test, we expect the error message to be logged. However, I don't think we verify logged messages in our other tests.

Reply:
> I agree, I was expecting to see a test that would cause `emailService.sendDatasetDeniedMessage()` to be called as that's the new code here.

The positive test already [exists in the test previous](https://github.com/DataBiosphere/consent/blob/8c6af8372e0ba4c37b673478c077db31572451b7/src/test/java/org/broadinstitute/consent/http/service/DatasetServiceTest.java#L742) to this one.